operational_analysis/methods/plant_analysis.py

```diff
@@ -714,37 +907,54 @@ def sample_long_term_reanalysis(self, n, r):
             1. The reanalysis product
             2. The number of years to use in the long-term correction
 
-        A memoized approach is used here since there are a finite combination of long-term wind speeds:
-            (3 reanalysis products x 10 different year combinations) = 30 total combinations
-
-        This memoized approach saves significant computaitonal time in the Monte Carlo simulation
-
         Args:
-           n(:obj:`integer`): The Monte Carlo Iteration Number
+           n(:obj:`integer`): The number of years for the windiness correction
            r(:obj:`string`): The reanalysis product used for Monte Carlo sample 'n'
 
         Returns:
-           :obj:`pandas.DataFrame`: the windiness-corrected or 'long-term' annualized monthly wind speeds
-
-        """
-        # Check if valid data has already been calculated and stored. If so, just return it
-        if (r, n) in self.long_term_sampling:
-            ws_monthly = self.long_term_sampling[(r, n)]
-            return ws_monthly
-
-        # If data hasn't yet been computed, peform the calculation
-        ws_df = self._reanalysis_monthly[r].to_frame().dropna()  # Drop NA values from monthly reanalysis data series
-        ws_data = ws_df.tail(n * 12)  # Get last 'x' years of data from reanalysis product
-        ws_monthly = ws_data.groupby(ws_data.index.month)[r].mean()  # Get long-term annualized monthly wind speeds
-
-        # Store result in dictionary
-        self.long_term_sampling[(r, n)] = ws_monthly
-
-        # Return result
-        return ws_monthly
+           :obj:`pandas.DataFrame`: the windiness-corrected or 'long-term' annualized monthly/daily wind speeds
+
+        """
+        ws_df = self._reanalysis_aggregate[r].to_frame().dropna()  # Drop NA values from monthly/daily reanalysis data series
+        ws_data = ws_df.tail(n * self._calendar_samples)  # Get last 'x' years of data from reanalysis product
+        ws_aggregate = self.groupby_time_res(ws_data)[r] # Get long-term annualized daily wind speeds
+        ws_aggregate.reset_index(drop=True, inplace=True)
+        # IAV calculation
+        ws_aggregate_sd = ws_data.groupby(ws_data.index.month)[r].std() # Get long-term annualized stdev of daily wind speeds            
+        if self.time_resolution == 'D':
+            num_days_lt = (31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31)
+            ws_aggregate_sd = np.repeat(ws_aggregate_sd, num_days_lt)
```

Comment:
This section now uses self.num_days_lt. However, since np.repeat requires integers, I have had to slightly make this line a bit more complicated:
`ws_aggregate_sd = np.repeat(ws_aggregate_sd, np.ceil(self.num_days_lt).astype(int))`